### PR TITLE
Refactor canvas single vite config

### PR DESCRIPTION
**Pull request by rhlin**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

内容：
1. 画布内的render单独构建成应用 base64内嵌到画布的lib构建产物中
2. 画布连接src开发的时候，引用一个esbuild external掉的插件，并且给画布内外分开设置importmap
3. 画布去掉内置tinyVue组件库的内置逻辑，bundle.json增加tinyVue组件的所有script和css

额外的内容：
1. 修复众多package.json缺包， 去掉devAlias能发现缺很多包
2. ~~devAlias缺toolbar/lock，补充回去~~
3. 物料bundle.json，env环境变量全部切到 unpkg.com （npm.onmicrosof.cn已经不可用，没有意义了）
4. 去掉了designer-demo build脚本后面的--mode 会导致读取不到变量

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Updated content delivery network (CDN) settings to enhance resource accessibility and loading times for both alpha and production environments.
	- Introduced new functions and configurations to streamline canvas initialization and import map management.
	- Added peer dependencies for enhanced localization and icon support across various packages.
	- Launched a new Vite plugin to manage entry points and asset handling in a modular fashion.
	- Restructured README documentation for the `tiny-engine-canvas` module, providing clearer guidance on architecture and build processes.

- **Bug Fixes**
	- Improved iframe rendering by using `srcdoc`, enhancing performance.

- **Chores**
	- Updated various dependency versions and configurations, ensuring compatibility with new features and improvements.

### CODE REVIEW

File: packages/build/vite-config/src/vite-plugins/genImportMapOnly.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+              injectTo: 'head-prepend'
+            },
+            ...(importMapStyles || []).map((url) => ({
+              tag: 'link',
+              attrs: {
+                rel: 'stylesheet',
+                href: url
+              },
+              injectTo: 'head-prepend'
+            }))
```
[the remark this answers is not in the document]
Comment (rhlin) [reply packages/build/vite-config/src/vite-plugins/genImportMapOnly.js:25]: currently dont need to validate url

File: packages/canvas/init-canvas/init-canvas.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import canvasScript from './canvas?url&build=canvas'
+import canvasHtml from './canvas.html?raw'
```
[the remark this answers is not in the document]
Comment (rhlin) [reply packages/canvas/init-canvas/init-canvas.js:2]: they are in file system， no need to  error handling

File: packages/canvas/init-canvas/init-canvas.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+export function initCanvas(importMap = {}, importStyleUrls = []) {
+  return {
+    html: canvasHtml
+      .replace('<!--%IMPORT_MAP%-->', `<script type="importmap">${JSON.stringify(importMap, null, 2)}</script>`)
+      .replace(
+        '<!--%IMPORT_STYLE%-->',
+        importStyleUrls.map((styleUrl) => `<link rel="stylesheet" crossorigin="" href="${styleUrl}">`).join('\n')
+      )
+      .replace('<!--%MAIN_SCRIPT%-->', `<script type="module" src="${canvasScript}"></script>`)
+  }
```
[the remark this answers is not in the document]
Comment (rhlin) [reply packages/canvas/init-canvas/init-canvas.js:13]: thx but it is not nessesary

File: packages/build/vite-config/src/canvas-dev-external.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  // 以下内容由于区块WebComponent加载需要补充
+  const blockRequire = {
+    externals: [/^@opentiny\/vue$/, /^@opentiny\/vue-icon$/],
+    imports: {
+      '@opentiny/vue': `${prefix}/node_modules/@opentiny/vue/runtime/tiny-vue.mjs`,
+      '@opentiny/vue-icon': `${prefix}/node_modules/@opentiny/vue/runtime/tiny-vue-icon.mjs`
+    },
+    importStyles: [`${prefix}/node_modules/@opentiny/vue-theme/index.css`]
+  }
+  // 以下内容由于物料协议不支持声明子依赖而@opentiny/vue需要依赖所以需要补充
```
Comment (chilingling) [inline packages/build/vite-config/src/canvas-dev-external.js:15]: 物料协议是不是可以先简单改成数组的形式？
```json
{
  "npm": {
    "script": [xxx, xxx1, xxx2],
    "css": [css1, css2, css3]
  }
}
```

File: packages/build/vite-config/src/vite-plugins/devAliasPlugin.js [hunk 5]
```diff
@@ -38,6 +38,7 @@ const getDevAlias = (useSourceAlias) => {
     '@opentiny/tiny-engine-toolbar-lang': path.resolve(basePath, 'packages/toolbars/lang/index.js'),
     '@opentiny/tiny-engine-toolbar-layout': path.resolve(basePath, 'packages/toolbars/layout/index.js'),
     '@opentiny/tiny-engine-toolbar-checkinout': path.resolve(basePath, 'packages/toolbars/lock/index.js'),
+    '@opentiny/tiny-engine-toolbar-lock': path.resolve(basePath, 'packages/toolbars/lock/index.js'),
```
Comment (chilingling) [inline packages/build/vite-config/src/vite-plugins/devAliasPlugin.js]: 这里  40 行就是锁的 alias

Comment (rhlin) [reply packages/build/vite-config/src/vite-plugins/devAliasPlugin.js]: fixed

File: packages/canvas/package.json [hunk 6]
```diff
@@ -31,6 +31,7 @@
     "@opentiny/tiny-engine-builtin-component": "workspace:*",
     "@opentiny/tiny-engine-meta-register": "workspace:*",
     "@opentiny/tiny-engine-i18n-host": "workspace:*",
+    "@opentiny/tiny-engine-plugin-materials": "workspace:*",
```
Comment (hexqi) [inline packages/canvas/package.json]: canvas不依赖materials包吧，我记得已经改成从注册表获取了，这个是多余的吧？

Comment (rhlin) [reply packages/canvas/package.json]: 这个是package.json代码格式化（参考第一个commit， 源文件在37行），merge后被保留了，已经确认没有引用，我这边去掉

File: packages/canvas/scripts/vite-plugin-separate-build.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        fileName: assets.fileName,
+        source: assets.code
+      })
+    }
+  })
+  return outputChunk
+}
+
+function isSameContent(a, b) {
+  if (typeof a === 'string') {
```
Comment (chilingling) [inline packages/canvas/scripts/vite-plugin-separate-build.js:55]: 这样是不是更简洁？（不确定是不是等效）
```javascript
const bufA = typeof a === 'string' ? Buffer.from(a) : a;

return Buffer.from(b).equals(a);
```

Comment (rhlin) [reply packages/canvas/scripts/vite-plugin-separate-build.js:55]: isSameContent 是参考 [workerPlugin](https://github.com/vitejs/vite/blob/main/packages/vite/src/node/plugins/worker.ts#L458) 写的, 也是一种单独构建方案里的，没有去变更，整个方案基本都是参考workerPlugin写的

这里的示例代码好像不等价 `bufA`定义后没有用过？

**Review by hexqi — DISMISSED** [review]

**Review by hexqi — APPROVED** [review]

**Pull request merged**